Set line position of data value flags

cari-lynn opened the pull request:
Why: In order to report better schema errors the source line is now saved

For all flags not read from a file the error is [reported](https://github.com/vmware-tanzu/carvel-ytt/pull/438/files#diff-2ab902ad23ffe96207ff6466403834df5538e509639024bfc503aed26a377cb8R562-R573) in the format `key=value`. 

Two cases I want to call out: 
### 1st case:
 When reporting an error from a data value read from a file eg: `--data-value-file foo=/path/file.yml` the line will be reported with only the file contents, and they key name and file name are reported at the top:
```
key 'foo' (key=file arg) dvs1.yml:
    |
  1 | not an integer
    |
```
Whereas other flags errors are reported exactly as they were input on the command line (eg: key=value). Please let me know if you have any feedback on this format.

### 2nd case: 
When targeting a library via `--data-value @lib:foo=4` the `@lib:` prefix is included in the error:
```
key 'foo' (kv arg):
         |
       1 | @lib:foo=42
```

cari-lynn committed:
Set position line of data value flags

- In order to report better schema errors the source line is now saved
- Applies to
  - `--data-value`
  - `--data-value-yaml`
  - `--data-value-env`
  - `--data-value-file`
  - `--data-value-env-yaml`

cari-lynn committed:
Include file name in `--data-value-file` error

- backfill test for `--data-values-file` which required no changes

pivotaljohn commented:
Good work, here, @cari-lynn.

As I was reading through the format of the error messages, I realized that there's some distance between the words the user employs to provide a data value on the command line and the words we use to describe them in an error message.

Playing around, I wonder what you think of this:

```
One or more data values were invalid
====================================

(data-value-file arg) foo=./dvs1.yml:
    |
  1 | not an integer
    |

    = found: string
    = expected: integer (by schema.yml:3)


(data-values-env arg) DVS:
    |
  1 | DVS_foo=not an integer
    |

    = found: string
    = expected: integer (by schema.yml:3)


(data-values arg):
    |
  1 | foo=42
    |

    = found: string
    = expected: integer (by schema.yml:3)


(data-values-yaml arg):
    |
  1 | foo=not an integer
    |

    = found: string
    = expected: integer (by schema.yml:3)
```

The latest two would be a little tricky, but I wonder if it made sense that `dataValuesFlagsSource` could include a `Name` which is the name of the source.

Thoughts?

cari-lynn committed:
Standardize data value flags errors

- Use flag name as error message
- Only exception is `--data-values-file which uses file name

cari-lynn commented:
Thanks @pivotaljohn. I like keeping the error messages wording consistent with the flag names 👍   
I also added the suggested change to `dataValuesFlagsSource`. Let me know what you think.

pivotaljohn reviewed (approved):
LGTM!
